pkg/splunk/enterprise/util.go

```diff
@@ -789,3 +790,38 @@ func setupAppInitContainers(client splcommon.ControllerClient, cr splcommon.Meta
 		}
 	}
 }
+
+// SetAppInfoStatus sets the last check interval to current time and
+// sets the flag to check the apps status in the next interval
+func SetAppInfoStatus(appInfoStatus *enterprisev1.AppInfoStatus) {
+	currentEpoch := time.Now().Unix()
+	appInfoStatus.LastAppInfoCheckTime = currentEpoch
+	appInfoStatus.NeedToCheckAfterPollInterval = true
+}
+
+func hasAppRepoCheckTimerExpired(appFrameworkRef enterprisev1.AppFrameworkSpec, appInfoStatus enterprisev1.AppInfoStatus) bool {
+	scopedLog := log.WithName("hasAppRepoCheckTimerExpired")
+	currentEpoch := time.Now().Unix()
+	scopedLog.Info("Checking if the app repo polling interval has expired", "LastAppInfoCheckTime", strconv.FormatInt(appInfoStatus.LastAppInfoCheckTime, 10), "current epoch time", strconv.FormatInt(currentEpoch, 10))
+	return appInfoStatus.LastAppInfoCheckTime+int64(appFrameworkRef.AppsRepoPollInterval) < currentEpoch
+}
+
+// ShouldCheckAppStatus checks whether we need to check the app status in the current reconcile
+func ShouldCheckAppStatus(appFrameworkRef enterprisev1.AppFrameworkSpec, appInfoStatus enterprisev1.AppInfoStatus) bool {
+	scopedLog := log.WithName("ShouldCheckAppStatus")
+
+	scopedLog.Info("Checking to see if we need to probe the status of apps on remote storage")
+
+	// check if its the first time or we are coming after appsRepoPollInterval secs.
+	return appInfoStatus.NeedToCheckAfterPollInterval == false ||
+		(appInfoStatus.NeedToCheckAfterPollInterval == true && hasAppRepoCheckTimerExpired(appFrameworkRef, appInfoStatus))
+}
```

Comment:
As mentioned earlier, just returning hasAppRepoCheckTimerExpired() will be simple and clear, as we are not differentiating the error condition for an 'apply' command vs. timer expiry.(?)